src/toolbars/DeviceToolBar.cpp

```diff
@@ -96,6 +139,34 @@ const DeviceToolBar &DeviceToolBar::Get( const AudacityProject &project )
    return Get( const_cast<AudacityProject&>( project )) ;
 }
 
+// This is a convenience function that allows for button creation in
+// MakeButtons() with fewer arguments
+AButton* DeviceToolBar::MakeButton(DeviceToolBar* pBar,
+   teBmps eEnabledUp, teBmps eEnabledDown, teBmps eDisabled,
+   int id,
+   bool processdownevents,
+   const TranslatableString& label)
+{
+   AButton* r = ToolBar::MakeButton(pBar,
+      bmpRecoloredUpLarge, bmpRecoloredDownLarge, bmpRecoloredUpHiliteLarge, bmpRecoloredHiliteLarge,
+      eEnabledUp, eEnabledDown, eDisabled,
+      wxWindowID(id),
+      wxDefaultPosition, processdownevents,
+      theTheme.ImageSize(bmpRecoloredUpLarge));
+   r->SetLabel(label);
+   enum {
+      deflation =
+#ifdef __WXMAC__
+      6
+#else
+      12
+#endif
+   };
+   r->SetFocusRect(r->GetClientRect().Deflate(deflation, deflation));
+
+   return r;
+}
```

Comment:
This looks like duplication from ControlToolBar.cpp.  Instead, let's lower the repeated function into ToolBar as an overload of MakeButton.